src/nifs/mod.rs

```diff
@@ -9,222 +9,58 @@
+    /// metadata for prover including hash of public params
+    type ProverParam;
 
-/// Cryptographic commitments to the [`CrossTerms`].
-pub type CrossTermCommits<C> = Vec<C>;
+    /// metadata for verifier including hash of public params
+    type VerifierParam;
 
-/// NIFS: Non Interactive Folding Scheme
-///
-/// Given a polynomial relation `P(x_1,...,x_n)` with polynomial degree `d.
-/// After folding two such (identical) relations, we have:
-/// - `P(x_1 + r*y_1, ..., x_n + r * y_n) = P(x_1, ..., x_n) + \sum_{k=1}^{d-1} T_k + r^d * P(y_1,...,y_n)`
-/// - `cross_term = [T_1,...,T_{d-1}]`
-/// - `cross_term_commits = [Comm(T_1),...,Comm(T_{d-1})]`
-///
-/// Please refer to: [notes](https://hackmd.io/@chaosma/BJvWmnw_h#31-NIFS)
-// TODO Replace links to either the documentation right here, or the official Snarkify resource
-#[derive(Clone, Debug)]
-pub struct NIFS<C: CurveAffine, RO: ROTrait<C>> {
-    pub(crate) cross_term_commits: CrossTermCommits<C>,
-    _marker: PhantomData<RO>,
-}
-
-impl<C: CurveAffine, RO: ROTrait<C>> NIFS<C, RO> {
-    /// Commits to the cross terms between two Plonk instance-witness pairs.
-    ///
-    /// This method calculates the cross terms and their commitments, which
-    /// are essential for the folding process.
-    ///
-    /// # Arguments
-    /// * `ck`: The commitment key.
-    /// * `S`: the Plonk structure shared by both instance-witness pairs.
-    /// * `U1`: The first relaxed Plonk instance.
-    /// * `W1`: The witness for the first relaxed Plonk instance.
-    /// * `U2`: The second Plonk instance.
-    /// * `W2`: The witness for the second Plonk instance.
-    ///
-    /// # Returns
-    /// A tuple containing the cross terms and their commitments.
-    ///
-    /// # Context
-    /// The cross terms are derived from the polynomial relations
-    /// of the two instance-witness pairs. They play a crucial role
-    /// in the folding process, allowing two polynomial relations
-    /// to be combined into one.
-    pub fn commit_cross_terms(
-        ck: &CommitmentKey<C>,
-        S: &PlonkStructure<C>,
-        U1: &RelaxedPlonkInstance<C>,
-        W1: &RelaxedPlonkWitness<C::ScalarExt>,
-        U2: &PlonkInstance<C>,
-        W2: &PlonkWitness<C::ScalarExt>,
-    ) -> Result<(CrossTerms<C>, CrossTermCommits<C>), Error> {
-        let offset = S.num_non_fold_vars();
-        let num_row = if !S.fixed_columns.is_empty() {
-            S.fixed_columns[0].len()
-        } else {
-            S.selectors[0].len()
-        };
-        let data = PlonkEvalDomain {
-            num_advice: S.num_advice_columns,
-            num_lookup: S.num_lookups(),
-            challenges: concat_vec!(&U1.challenges, &[U1.u], &U2.challenges, &[U2.to_relax().u]),
-            selectors: &S.selectors,
-            fixed: &S.fixed_columns,
-            W1s: &W1.W,
-            W2s: &W2.W,
-        };
+    /// Accumulator contains AccumulatorInstance and AccumulatorWitness (e.g. RelaxedPlonkWitness)
```

Comment:
If you referense to some existing types - use rustdoc https://doc.rust-lang.org/rustdoc/write-documentation/linking-to-items-by-name.html

Reply:
This is important because it gives the impression that some types exist and you look for them by hand (for example here `AccumulatorWitness`) and they don't exist and it can be confusing

Reply:
good point